### PR TITLE
Fixed z-index and added autohide on window resize

### PR DESCRIPTION
**Pull request by grishat**

### CODE REVIEW

File: BlazorContextMenu/Components/ContextMenuBase.cs [hunk 1]
```diff
@@ -108,6 +108,12 @@ public abstract class ContextMenuBase : MenuTreeComponent
         [Parameter]
         public bool AutoHide { get; set; } = true;
 
+        /// <summary>
+        /// Set CSS z-index for overlapping other html elements. Default: 1000
+        /// </summary>
+        [Parameter]
+        public int Z { get; set; } = 1000;
```
Comment (stavroskasidis) [inline BlazorContextMenu/Components/ContextMenuBase.cs:115]:
```suggestion
        public int ZIndex { get; set; } = 1000;
```

File: BlazorContextMenu/Components/ContextMenu.razor [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿@inherits ContextMenuBase
 @namespace BlazorContextMenu
 
-<div @attributes="@Attributes" class="@($"{BaseClass} {ClassCalc} {DisplayClassCalc}")" id="@Id" style="@($"left:{X}px;top:{Y}px;")" data-autohide="@AutoHide.ToString().ToLower()">
+<div @attributes="@Attributes" class="@($"{BaseClass} {ClassCalc} {DisplayClassCalc}")" id="@Id" style="@($"left:{X}px;top:{Y}px;z-index:{Z};")" data-autohide="@AutoHide.ToString().ToLower()">
```
Comment (stavroskasidis) [inline BlazorContextMenu/Components/ContextMenu.razor:4]:
```suggestion
<div @attributes="@Attributes" class="@($"{BaseClass} {ClassCalc} {DisplayClassCalc}")" id="@Id" style="@($"left:{X}px;top:{Y}px;z-index:{ZIndex};")" data-autohide="@AutoHide.ToString().ToLower()">
```

**Comment by stavroskasidis** [comment]
Nice changes, thanks for submitting them. I added a minor suggestion for the z-index parameter

**Comment by grishat** [comment]
And what about window resize?

**Comment by stavroskasidis** [comment]
I didn't think about window resizing, I agree with the change

**Comment by stavroskasidis** [comment]
Should I commit my suggestions ?

**Comment by grishat** [comment]
Of course.

**Pull request merged**